### PR TITLE
	improve the performance of fs.Open() by unzipping in fs.New() and simplifying http.File wrapper

### PR DESCRIPTION
**Pull request by matthewdale**
Please consider reviewing and merging https://github.com/rakyll/statik/pull/22 before this PR. I'd like to rebase this branch after that PR is merged to take advantage of the additional test coverage.

Benchmark results (from benchmark in https://github.com/rakyll/statik/pull/22):
master:
```
$ go test -v -bench=. -run=^$ github.com/rakyll/statik/fs
BenchmarkOpen-4   	  500000	      2398 ns/op
PASS
ok  	github.com/rakyll/statik/fs	1.237s
```
this PR:
```
$ go test -v -bench=. -run=^$ github.com/rakyll/statik/fs
BenchmarkOpen-4   	10000000	       137 ns/op
PASS
ok  	github.com/rakyll/statik/fs	1.526s
```

### CODE REVIEW

File: fs/fs.go [hunk 1]
```diff
@@ -40,107 +45,105 @@ func Register(data string) {
 }
 
 // New creates a new file system with the registered zip contents data.
+// It unzips all files and stores them in an in-memory map.
 func New() (http.FileSystem, error) {
 	if zipData == "" {
-		return nil, errors.New("statik/fs: No zip data registered.")
+		return nil, errors.New("statik/fs: No zip data registered")
```
Comment (rakyll) [inline fs/fs.go]: could you fix the format of the error message while you are there?

return nil, errors.New("statik/fs: no zip data registered")

File: fs/fs.go [hunk 2]
```diff
@@ -40,107 +45,105 @@ func Register(data string) {
...
 		return nil, err
 	}
-	files := make(map[string]*zip.File)
-	for _, file := range zipReader.File {
-		files["/"+file.Name] = file
+	files := make(map[string]file)
+	for _, zipFile := range zipReader.File {
+		unzipped, err := unzip(zipFile)
+		if err != nil {
+			return nil, fmt.Errorf(
```
Comment (rakyll) [inline fs/fs.go]: fitting these errorfs to one line would be great.

return nil, fmt.Errorf("static/fs: error unzipping file..."

File: fs/fs.go (hunk 1 above)
Comment (matthewdale) [inline fs/fs.go]: Updated.

File: fs/fs.go (hunk 2 above)
Comment (matthewdale) [inline fs/fs.go]: Changed to a single line.

**Comment by matthewdale** [comment]
@rakyll I have read and signed a Google CLA.

**Comment by rakyll** [comment]
Thanks!

**Pull request merged**